fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1109,7 +1114,10 @@ struct CreateTenantImpl {
 				// The tenant creation has already started, so resume where we left off
 				self->tenantEntry = existingEntry.get();
 				ASSERT(existingEntry.get().assignedCluster.present());
-
+				if (self->preferAssignedCluster &&
+				    existingEntry.get().assignedCluster.get() != self->tenantEntry.assignedCluster.get()) {
+					throw cluster_not_found();
```

Comment:
This should throw something like `invalid_tenant_configuration`, since `cluster_not_found` would mean that your chosen cluster doesn't exist. I'd also add a trace event to this case so that we can look at the logs and get more details.